apps/game_backend/lib/game_backend/users.ex

```diff
@@ -74,60 +70,104 @@ defmodule GameBackend.Users do
       {:error, :not_found}
   """
   def get_user_by_id_and_game_id(id, game_id) do
+    quest_refresh_at =
+      NaiveDateTime.utc_now()
+      |> NaiveDateTime.add(1, :day)
+      |> NaiveDateTime.beginning_of_day()
+      |> NaiveDateTime.to_iso8601()
+
     q =
       from(u in User,
         where: u.id == ^id and u.game_id == ^game_id,
         join: unit in Unit,
         on: u.id == unit.user_id,
         preload: [units: [:character, :items], currencies: :currency],
         group_by: u.id,
-        select: %{u | prestige: sum(unit.prestige)}
+        select: %{u | prestige: sum(unit.prestige), quest_refresh_at: ^quest_refresh_at}
       )
+      |> quests_preloads()
+      |> arena_match_results_preloads()
+
+    user =
+      Repo.one(q)
+      |> add_quest_progress_and_goal()
```

Comment:
This will get time consuming. Is this a decision made to send the whole user information (quests, etc) at once in a single endpoint that calls this query?

Reply:
yup this will send the quest information in the same endpoint. It's debatable to add it in the same endpoint or do a new specific endpoint for the quests. What do you think?

Reply:
Discussed in a call. We will leave it as it is until this is a priority.